ex_app/lib/main.py

```diff
@@ -502,8 +502,96 @@ def delete_listener(registered_listener: dict) -> bool:
     return r
 
 
+def create_nextcloud_auth_variable() -> bool:
+    r = httpx.post(
+        url="http://127.0.0.1:8000/api/w/nextcloud/variables/create",
+        cookies={"token": USERS_STORAGE["admin@windmill.dev"]["token"]},
+        json={
+            "path": "u/admin/exapp_token",
+            "value": os.environ["APP_SECRET"],
+            "is_secret": True,
+            "is_oauth": False,
+            "description": "ExApp Authentication Variable",
+        }
+    )
+    if r.status_code >= 400:
+        LOGGER.critical("Can not create Nextcloud Auth Variable: %s %s", r.status_code, r.text)
+        return False
+    return True
+
+
+def update_nextcloud_auth_variable() -> bool:
+    r = httpx.post(
+        url="http://127.0.0.1:8000/api/w/nextcloud/variables/update/u/admin/exapp_token",
+        cookies={"token": USERS_STORAGE["admin@windmill.dev"]["token"]},
+        json={"value": os.environ["APP_SECRET"]}
+    )
+    if r.status_code >= 400:
+        LOGGER.critical("Can not update Nextcloud Auth Variable: %s %s", r.status_code, r.text)
+        return False
+    return True
+
+
+def create_nextcloud_auth_resource() -> bool:
+    r = httpx.post(
+        url="http://127.0.0.1:8000/api/w/nextcloud/resources/create",
+        cookies={"token": USERS_STORAGE["admin@windmill.dev"]["token"]},
+        json={
+            "path": "u/admin/exapp_resource",
+            "resource_type": "nextcloud",
+            "baseUrl": "unknown",
```

Comment:
Ah, damn. This would be good to have :S

Reply:
> Ah, damn. This would be good to have :S

we can add API to AppAPI to get value of `overwrite.cli.url` - will it be enough?

Reply:
anyway when AppAPI will provide such API we will do another PR